backend/backend/views.py

```diff
@@ -17,10 +17,13 @@ def post(self, request, *args, **kwargs):
                                            context={'request': request})
         serializer.is_valid(raise_exception=True)
         user = serializer.validated_data['user']
+
         token, created = Token.objects.get_or_create(user=user)
 
-        # token_expire_handler will check, if the token is expired it will generate new one
-        is_expired, token = token_expire_handler(token)
+        # token should be new each time, remove the old one
+        if not created:
+            token.delete()
+            token = Token.objects.create(user=token.user)
```

Comment:
What about trying to delete it if it exists and then create it in all cases? 
It seems easier to maintain: creation is done at a single place.

```
# token should be new each time, so remove old one if it exists
Token.objects.filter(user=user).delete()
token = Token.objects.create(user=user)
```
(this code has not been tested :))